src/App.tsx

```diff
@@ -1,30 +1,86 @@
+/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { getTodos } from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { TodoList } from './components/TodoList/TodoList';
+import { ErrorNotification } from './components/ErrorNotification/ErrorNotification';
+import { Footer } from './components/Footer/Footer';
+import { FilterBy } from './types/FilterBy';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState(false);
+  const [filterBy, setFilterBy] = useState('all');
+
+  const filterTodos = todos.filter(todo => {
+    switch (filterBy) {
+      case FilterBy.All:
+        return todo;
+      case FilterBy.Active:
+        return !todo.completed;
+      case FilterBy.Completed:
+        return todo.completed;
+      default:
+        return 0;
+    }
+  });
+
+  let userId = 0;
+
+  if (user?.id) {
+    userId = user.id;
+  }
+
+  userId = 5; /* example */
+
+  if (error) {
+    setTimeout(() => {
+      setError(false);
+    }, 3000);
+  }
 
   useEffect(() => {
     // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+
+    getTodos(userId)
+      .then(setTodos)
+      .catch(() => setError(true));
```

Comment:
Let’s use async await. It gives you a simpler way to work with asynchronous promise-based code